In [1]:
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

## 1. 공공데이터포털 데이터 취합

In [2]:
data_go = pd.read_csv('data_go_kr_data.csv', index_col=0)
print(data_go.shape)
data_go.head(1)

(37475, 24)


서비스                    분류체계      제공기관  \
0  충청남도 당진시_산업단지 입주기업 현황_20220304  산업·통상·중소기업 - 산업·중소기업일반  충청남도 당진시   

          관리기관  관리기관 전화번호 보유근거 수집방법 업데이트 주기   차기 등록 예정일 매체유형  ...  \
0  공공데이터활용지원센터  1566-0025  NaN  NaN      연간  2023-09-29  텍스트  ...   

                키워드          등록          수정                    제공형태  \
0  산업단지,산업단지입주기업,산업  2022-09-29  2022-10-07  공공데이터포털에서 다운로드(원문파일등록)   

                                                  설명  \
0  충청남도 당진시 산업단지 입주 기업 현황입니다.(연번, 단지명, 회사명, 업종명, ...   

                      기타 유의사항 비용부과유무 비용부과기준 및 단위  \
0  개인정보 포함 된 정보는 공람 처리 하였습니다.     무료          건    

                                 이용허락범위  \
0  \n\n   \t\t\t이용허락범위 제한 없음\n   \t\t\n   

                                                url  
0  https://www.data.go.kr/data/15106898/fileData.do  

[1 rows x 24 columns]

In [3]:
data_go = data_go[['서비스', '설명', 'url', '매체유형', '제공기관', '분류체계']]
data_go['ori_source'] = '공공데이터포털'
data_go.columns = ['title', 'description', 'url', 'data_type', 'source', 'ori_label', 'ori_source']

data_go['ori_label'] = data_go['ori_label'].str.extract('([\S]+)').iloc[:, 0]
data_go['label'] = data_go.replace({'ori_label' : {
    '산업·통상·중소기업': '제조소비',
    '농림': '식품건강',
    '보건': '보건의료',
    '재정·세제·금융': '경제금융',
    '일반공공행정': '행정법률',
    '사회복지': '사회복지',
    '환경': '환경기상',
    '교통및물류': '교통물류',
    '문화체육관광': '문화관광',
    '해양수산': '식품건강',
    '공공질서및안전': '재난안전',
    '지역개발': '주택토지',
    '통신': '교육과학',
    '교육': '교육과학',
    '과학기술': '교육과학',
    '통일·외교': '통일외교안보',
    '국방': '통일외교안보'
}})['ori_label']
data_go

title  \
0         충청남도 당진시_산업단지 입주기업 현황_20220304   
1              전라남도 강진군_미곡처리장현황_20221006   
2        경상남도 거창군_월별 코로나 확진자 현황_20221007   
3            충청남도 부여군_1인당지방세부담액_20221006   
4            전라남도 보성군_담배소매인지정현황_20221006   
...                                  ...   
37470            암검진기관 현황 (2014.3.21 기준)   
37471     처인구 물가안정모범업소 현황 (2014.3.22 기준)   
37472  공중위생(목욕업)소 인허가 현황 (2013.12.31 기준)   
37473      가정상비의약품 판매소 현황 (2014.3.21 기준)   
37474           식품제조가공업 현황 (2014.3.4 기준)   

                                             description  \
0      충청남도 당진시 산업단지 입주 기업 현황입니다.(연번, 단지명, 회사명, 업종명, ...   
1      전라남도 강진군 미곡처리장 현황에 대한 데이터로소 시군, 명칭, 소재지, 관리부서,...   
2      경상남도 거창군 월별 코로나19 관련 확진자 및 사망자 수 현황 데이터로 월별 확진...   
3      상기 데이터는 연도별 주민 1인당 또는 세대 당 부담된  지방세액에 대한 정보로 조...   
4      전라남도 보성군 담배소매인지정현황에 관한 공공데이터로 업소명, 지번주소, 도로명주소...   
...                                                  ...   
37470                            용인시 수지구 관내 암검진기관 정보입니다.   
37471                         용인시 처인구 관내 물가안정모범업소 정보입니다.   
37472                              용인시 관내 목욕업 인허가 정보입니다.   
37473                         용인시 관내 안전상비의약품 판매업소 정보입니다.   
37474                            용인시 관내 식품제조 가공업체 정보입니다.   

                                                    url data_type    source  \
0      https://www.data.go.kr/data/15106898/fileData.do       텍스트  충청남도 당진시   
1      https://www.data.go.kr/data/15106993/fileData.do       텍스트  전라남도 강진군   
2      https://www.data.go.kr/data/15098865/fileData.do       텍스트  경상남도 거창군   
3      https://www.data.go.kr/data/15080007/fileData.do       텍스트  충청남도 부여군   
4      https://www.data.go.kr/data/15035564/fileData.do       텍스트  전라남도 보성군   
...                                                 ...       ...       ...   
37470  https://www.data.go.kr/data/15050606/fileData.do       텍스트   경기도 용인시   
37471  https://www.data.go.kr/data/15053858/fileData.do       텍스트   경기도 용인시   
37472   https://www.data.go.kr/data/3072096/fileData.do       텍스트   경기도 용인시   
37473   https://www.data.go.kr/data/3062985/fileData.do       텍스트   경기도 용인시   
37474   https://www.data.go.kr/data/3063528/fileData.do       텍스트   경기도 용인시   

        ori_label ori_source label  
0      산업·통상·중소기업    공공데이터포털  제조소비  
1              농림    공공데이터포털  식품건강  
2              보건    공공데이터포털  보건의료  
3        재정·세제·금융    공공데이터포털  경제금융  
4          일반공공행정    공공데이터포털  행정법률  
...           ...        ...   ...  
37470          보건    공공데이터포털  보건의료  
37471      일반공공행정    공공데이터포털  행정법률  
37472      일반공공행정    공공데이터포털  행정법률  
37473          보건    공공데이터포털  보건의료  
37474          보건    공공데이터포털  보건의료  

[37475 rows x 8 columns]

## 2. 빅데이터 지도 데이터 취합

In [4]:
bigdata_map = pd.read_csv('bigdata_map_data.csv')
print(bigdata_map.shape)
bigdata_map.head(1)

(83692, 53)


id     type                         title  \
0  22424650  dataset  문화체육관광부 국립중앙박물관_국립부여박물관 유물정보   

                                                desc      identifier license  \
0  국립부여박물관 유물정보DB입니다. '유물명, 유물번호, 국적/시대, 재질, 출토지,...  datagov3078546     NaN   

  access_rights language                                     landing_page  \
0        public       ko  https://www.data.go.kr/data/3078546/fileData.do   

  rights  ... large_category medium_category ispublished data_type  \
0    NaN  ...           문화체육             NaN           1        정형   

                  keywords  file_extensions  topics  dn_flag dn_down_url  \
0  ['국립부여박물관', '유물', '검색']          ['CSV']     NaN      0.0         NaN   

   dn_sample_chk  
0            NaN  

[1 rows x 53 columns]

In [5]:
bigdata_map = bigdata_map[['title', 'desc', 'landing_page', 'file_extensions', 'platform_title', 'large_category']]
bigdata_map['ori_source'] = '빅데이터지도'
bigdata_map['file_extensions'] = bigdata_map['file_extensions'].str.replace('[', '').str.replace(']', '').str.replace("'", '')
bigdata_map.columns = ['title', 'description', 'url', 'data_type', 'source', 'ori_label', 'ori_source']

bigdata_map['label'] = bigdata_map.replace({'ori_label' : {
    '문화체육': '문화관광',
    '교육과학': '교육과학',
    '교통통신': '교통물류',
    '경제금융': '경제금융',
    '건설에너지': '주택토지',
    '농수산식품': '식품건강',
    '제조소비': '제조소비',
    '재난안전': '재난안전',
    '환경자원': '환경기상',
    '행정법률': '행정법률'
}})['ori_label']
bigdata_map

title  \
0                           문화체육관광부 국립중앙박물관_국립부여박물관 유물정보   
1                                       경기도 성남시_체력단련장_현황   
2                                     경기도 부천시_민간체육시설업 현황   
3                                      부산광역시_서구_노래연습장 정보   
4                                       경기도 고양시_직장운동부 현황   
...                                                  ...   
83687                                       SNS 경기도북부웹문서   
83688                                               공공정보   
83689                                              공정위정보   
83690                                             행정처분정보   
83691  2020년 행정안전부에서 전국민 대상으로 지급한 긴급재난지원금 경기지역화폐 일간 지...   

                                             description  \
0      국립부여박물관 유물정보DB입니다. '유물명, 유물번호, 국적/시대, 재질, 출토지,...   
1      성남시 내 위치한 체력단련장 데이터로, 구별, 동별, 상호명, 소재지도로명주소, 연...   
2      부천시 관내의 민간 체육시설 현황으로 상호명, 소재지 주소지(도로명), 소재지 전화...   
3      부산광역시 서구 관내 노래연습장 정보(상호명, 주소, 연락처, 시설면적, 등록일자 ...   
4      경기도 고양시 직장운동부 현황에 대한 데이터로 경기도 고양시 직장운동부의 종목, 창...   
...                                                  ...   
83687  주요 포털 및 SNS커뮤니티에서 경기도 북부 지역명으로 수집된 비정형 텍스트 데이터...   
83688   -  신용정보원의 기업별 공공기록정보(국세, 지방세 체납 등)를 지역별, 업종별,...   
83689   -  공정위원회 등록된 정보를 수집 및 전처리 가공하여, 기업 식별화 작업 후 통...   
83690   -  행정처분 현황 정보를 행정처분구분에 따라 가공 통계화한 시계열에 따른 업종별...   
83691  2020년 경기지역화폐 일반/정책발행카드의 비식별 개인별 행안부 긴급재난지원금 지급...   

                                                     url data_type   source  \
0        https://www.data.go.kr/data/3078546/fileData.do       CSV  공공데이터포털   
1        https://www.data.go.kr/data/3078606/fileData.do       CSV  공공데이터포털   
2        https://www.data.go.kr/data/3078656/fileData.do       CSV  공공데이터포털   
3        https://www.data.go.kr/data/3078672/fileData.do       CSV  공공데이터포털   
4        https://www.data.go.kr/data/3078722/fileData.do       CSV  공공데이터포털   
...                                                  ...       ...      ...   
83687  https://bigdata-region.kr/#/dataset/c5f466c7-9...       CSV     지역경제   
83688  https://www.bigdata-region.kr/#/dataset/fa48be...       CSV     지역경제   
83689  https://www.bigdata-region.kr/#/dataset/916733...       CSV     지역경제   
83690  https://www.bigdata-region.kr/#/dataset/f097a9...       CSV     지역경제   
83691  https://bigdata-region.kr/#/dataset/3d97eb1c-7...       CSV     지역경제   

      ori_label ori_source label  
0          문화체육     빅데이터지도  문화관광  
1          문화체육     빅데이터지도  문화관광  
2          문화체육     빅데이터지도  문화관광  
3          문화체육     빅데이터지도  문화관광  
4          문화체육     빅데이터지도  문화관광  
...         ...        ...   ...  
83687      행정법률     빅데이터지도  행정법률  
83688      행정법률     빅데이터지도  행정법률  
83689      행정법률     빅데이터지도  행정법률  
83690      행정법률     빅데이터지도  행정법률  
83691      행정법률     빅데이터지도  행정법률  

[83692 rows x 8 columns]

## 3. AI-hub 데이터 취합

In [6]:
ai_hub = pd.read_csv('ai_hub_data.csv', index_col=0)
print(ai_hub.shape)
ai_hub.head(1)

(384, 15)


일자       변경내용  비고  \
0  2022.7.29  콘텐츠 최초 등록 NaN   

                                                  소개  \
0  장애인, 노인, 어린이 등 교통약자의 이동평등권 보장을 위한 지능정보기술을 활용한 ...   

                                                구축목적 데이터 영역    데이터 유형  \
0  – 장애인의 이동평등권 보장 등 사회안전 분야에서 이동 편의성 확보를 위한 인공지능...  교통물류   3D , 이미지   

     데이터 형식 데이터 출처                                             라벨링 유형 라벨링 형식  \
0  jpg, pcd   직접취득  jpg: 2D Polygon Segmentation,pcd: 3D Bounding ...   JSON   

                                          데이터 활용 서비스  \
0  자율주행 휠체어 위험인지 기반 경로 예측 서비스, 강화학습 기반의 라이다 Point...   

                      데이터 구축년도/데이터 구축량  \
0  2021년/Image 288,000 , Lidar 144,000   

                                                 url  \
0  https://www.aihub.or.kr/aihubdata/data/view.do...   

                       title  
0  장애인 길안내 자율주행 휠체어 융합센서 데이터

In [7]:
ai_hub = ai_hub[['title', '소개', 'url', '데이터 유형', '데이터 출처', '데이터 영역']]
ai_hub['ori_source'] = 'AI_Hub'
ai_hub.columns = ['title', 'description', 'url', 'data_type', 'source', 'ori_label', 'ori_source']

ai_hub['label'] = ai_hub.replace({'ori_label' : {
    '교통물류 ': '교통물류',
    '한국어 ': '기타',
    '농축수산 ': '식품건강',
    '영상이미지 ': '기타',
    '헬스케어 ': '보건의료',
    '재난안전환경 ': '재난안전',
    '데이터 영역': '기타',
    '텍스트': '기타',
    '이미지': '기타',
    '오디오 , 비디오': '기타',
    '오디오 , 텍스트': '기타'
}})['ori_label']
ai_hub

title  \
0        장애인 길안내 자율주행 휠체어 융합센서 데이터   
1     일상생활 및 구어체 한-영 번역 병렬 말뭉치 데이터   
2         기술과학 분야 한영 번역 병렬 말뭉치 데이터   
3         전문분야 영-한·중-한 번역 말뭉치 (식품)   
4    방송 콘텐츠 한-중, 한-일 번역 병렬 말뭉치 데이터   
..                             ...   
380                       특허 지식베이스   
381                       법률 지식베이스   
382                       관광 지식베이스   
383                           일반상식   
384                     한국인 안면 이미지   

                                           description  \
0    장애인, 노인, 어린이 등 교통약자의 이동평등권 보장을 위한 지능정보기술을 활용한 ...   
1    • 신경망 기반 기계 번역기 학습 데이터로 활용하기 위한 한영, 영한 말뭉치\n• ...   
2    • 신경망 기반 기계 번역기 학습 데이터로 활용하기 위한 한영, 영한 말뭉치\n• ...   
3    • 신경망 기반 기계 번역기 학습 데이터로 활용하기 위한 영한, 중한 말뭉치\n• ...   
4    • 신경망 기반 기계 번역기 학습 데이터로 활용하기 위한 한중, 한일 말뭉치\n• ...   
..                                                 ...   
380  전기전자 분야 특허, 실용 신안 100만 개를 대상으로 특허 청구항에 구조화 정보 ...   
381  국민 생활과 밀접한 분야의 법령, 조문, 판례, 법률 상담 데이터와 법령, 판례, ...   
382  식당 주변 전경(간판, 입간판, 매장 전경, 메뉴판, 표지판) 사진을 포함한 150...   
383  한국어 위키백과 내 주요 문서 15만 개에 포함된 지식을 추출하여 객체(entity...   
384  한국인 얼굴 특성에 맞는 각도, 조도, 가림, 표정, 해상도 등으로 한국형 얼굴 데...   

                                                   url data_type source  \
0    https://www.aihub.or.kr/aihubdata/data/view.do...  3D , 이미지   직접취득   
1    https://www.aihub.or.kr/aihubdata/data/view.do...       텍스트    학술지   
2    https://www.aihub.or.kr/aihubdata/data/view.do...       텍스트    학술지   
3    https://www.aihub.or.kr/aihubdata/data/view.do...       텍스트    학술지   
4    https://www.aihub.or.kr/aihubdata/data/view.do...       텍스트    학술지   
..                                                 ...       ...    ...   
380  https://www.aihub.or.kr/aihubdata/data/view.do...       NaN    NaN   
381  https://www.aihub.or.kr/aihubdata/data/view.do...       NaN    NaN   
382  https://www.aihub.or.kr/aihubdata/data/view.do...       NaN    NaN   
383  https://www.aihub.or.kr/aihubdata/data/view.do...       NaN    NaN   
384  https://www.aihub.or.kr/aihubdata/data/view.do...       이미지    NaN   

    ori_label ori_source label  
0       교통물류      AI_Hub  교통물류  
1        한국어      AI_Hub    기타  
2        한국어      AI_Hub    기타  
3        한국어      AI_Hub    기타  
4        한국어      AI_Hub    기타  
..        ...        ...   ...  
380       텍스트     AI_Hub    기타  
381       텍스트     AI_Hub    기타  
382       이미지     AI_Hub    기타  
383       텍스트     AI_Hub    기타  
384    영상이미지      AI_Hub    기타  

[384 rows x 8 columns]

## 4. 서울 열린데이터 광장 데이터 취합

In [8]:
data_seoul = pd.read_csv('data_seoul.csv', index_col=0)
print(data_seoul.shape)
data_seoul['source'] = data_seoul['제공기관'] + ' ' + data_seoul['제공부서']
data_seoul.head(1)

(5500, 9)


카테고리                                                URL           제공형식  \
0   교통  https://data.seoul.go.kr/dataList/OA-15443/S/1...  SHEET,OpenAPI   

                     이름                                                 명세  \
0  서울시 남산 혼잡통행료 감면내역 정보  서울특별시 남산 혼잡통행료 감면내역 정보입니다.\n감면일자, 감면지역, 감면도로, ...   

    제공기관           제공부서         수정일                관련 태그               source  
0  서울특별시  기후환경본부 친환경차량과  2022-10-17  전자태그,1호터널,3호터널,남산터널  서울특별시 기후환경본부 친환경차량과

In [9]:
data_seoul = data_seoul[['이름', '명세', 'URL', '제공형식', 'source', '카테고리']]
data_seoul['ori_source'] = '서울데이터광장'
data_seoul.columns = ['title', 'description', 'url', 'data_type', 'source', 'ori_label', 'ori_source']

data_seoul['label'] = data_seoul.replace({'ori_label' : {
    '교통': '교통물류',
    '복지': '사회복지',
    '주택/건설': '주택토지',
    '도시관리': '주택토지',
    '환경': '환경기상',
    '일반행정': '행정법률',
    '안전': '재난안전',
    '보건': '보건의료',
    '문화/관광': '문화관광',
    '산업/경제': '경제금융',
    '교육': '교육과학'
}})['ori_label']
data_seoul

title  \
0          서울시 남산 혼잡통행료 감면내역 정보   
1                  서울글로벌센터 구인광고   
2            서울시 서울형 희망의 집수리 정보   
3     서울글로벌센터 월별 접수유형별 분류별 상담실적   
4               서울글로벌센터 월별 상담실적   
...                         ...   
5495    서울시 구로구 식품첨가물제조업 인허가 정보   
5496     서울시 송파구 국제회의시설업 인허가 정보   
5497      서울시 중구 국제회의시설업 인허가 정보   
5498     서울시 강남구 국제회의시설업 인허가 정보   
5499   서울시 송파구 비디오물시청제공업 인허가 정보   

                                            description  \
0     서울특별시 남산 혼잡통행료 감면내역 정보입니다.\n감면일자, 감면지역, 감면도로, ...   
1     사업자등록증이 있는 사업장에서 외국인 대상으로 구인광고를 낼 수 있는 서비스입니다....   
2     서울형 희망의 집수리 사업 대상자가 거주하는 행정구역, 주택유형, 주거형태, 방갯수...   
3                    서울글로벌센터 월별 상담실적을 접수유형별/분류별로 제공합니다.   
4     서울글로벌센터에서 상담 받은 내용 중 유형에 따른 전화, 방문, 이메일, 게시판 이...   
...                                                 ...   
5495  구로구의 식품을 제조 또는 가공할 때 기호가치를 향상시키거나 영양가치를 높일목적의 ...   
5496  송파구의 컨벤션센터 등 대규모 관광 수요를 유발하는 국제회의를 개최할 수 있는 시설...   
5497  중구의 컨벤션센터 등 대규모 관광 수요를 유발하는 국제회의를 개최할 수 있는 시설을...   
5498  강남구의 컨벤션센터 등 대규모 관광 수요를 유발하는 국제회의를 개최할 수 있는 시설...   
5499  송파구의 숙박 및 휴게 등의 목적으로 이용하는 장소 또는 시설에서 비디오물 시청기자...   

                                                    url      data_type  \
0     https://data.seoul.go.kr/dataList/OA-15443/S/1...  SHEET,OpenAPI   
1     https://data.seoul.go.kr/dataList/OA-15732/S/1...  SHEET,OpenAPI   
2     https://data.seoul.go.kr/dataList/OA-15445/S/1...  SHEET,OpenAPI   
3     https://data.seoul.go.kr/dataList/OA-15739/S/1...  SHEET,OpenAPI   
4     https://data.seoul.go.kr/dataList/OA-15735/S/1...  SHEET,OpenAPI   
...                                                 ...            ...   
5495  https://data.seoul.go.kr/dataList/OA-18293/S/1...  SHEET,OpenAPI   
5496  https://data.seoul.go.kr/dataList/OA-16990/S/1...  SHEET,OpenAPI   
5497  https://data.seoul.go.kr/dataList/OA-16968/S/1...  SHEET,OpenAPI   
5498  https://data.seoul.go.kr/dataList/OA-16989/S/1...  SHEET,OpenAPI   
5499  https://data.seoul.go.kr/dataList/OA-17365/S/1...  SHEET,OpenAPI   

                         source ori_label ori_source label  
0           서울특별시 기후환경본부 친환경차량과        교통    서울데이터광장  교통물류  
1        서울특별시 여성가족정책실 가족다문화담당관        복지    서울데이터광장  사회복지  
2     서울특별시 주택정책실 주택공급기획관 주택정책과     주택/건설    서울데이터광장  주택토지  
3        서울특별시 여성가족정책실 가족다문화담당관        복지    서울데이터광장  사회복지  
4        서울특별시 여성가족정책실 가족다문화담당관        복지    서울데이터광장  사회복지  
...                         ...       ...        ...   ...  
5495       구로구 스마트도시정책관 빅데이터담당관        보건    서울데이터광장  보건의료  
5496       송파구 스마트도시정책관 빅데이터담당관     문화/관광    서울데이터광장  문화관광  
5497        중구 스마트도시정책관 빅데이터담당관     문화/관광    서울데이터광장  문화관광  
5498       강남구 스마트도시정책관 빅데이터담당관     문화/관광    서울데이터광장  문화관광  
5499       송파구 스마트도시정책관 빅데이터담당관     문화/관광    서울데이터광장  문화관광  

[5500 rows x 8 columns]

## 5. 취합한 데이터를 하나로 만들기

In [10]:
total_df = pd.concat([data_go, bigdata_map, ai_hub, data_seoul], axis=0).reset_index(drop=True)
total_df

title  \
0        충청남도 당진시_산업단지 입주기업 현황_20220304   
1             전라남도 강진군_미곡처리장현황_20221006   
2       경상남도 거창군_월별 코로나 확진자 현황_20221007   
3           충청남도 부여군_1인당지방세부담액_20221006   
4           전라남도 보성군_담배소매인지정현황_20221006   
...                                 ...   
127046          서울시 구로구 식품첨가물제조업 인허가 정보   
127047           서울시 송파구 국제회의시설업 인허가 정보   
127048            서울시 중구 국제회의시설업 인허가 정보   
127049           서울시 강남구 국제회의시설업 인허가 정보   
127050         서울시 송파구 비디오물시청제공업 인허가 정보   

                                              description  \
0       충청남도 당진시 산업단지 입주 기업 현황입니다.(연번, 단지명, 회사명, 업종명, ...   
1       전라남도 강진군 미곡처리장 현황에 대한 데이터로소 시군, 명칭, 소재지, 관리부서,...   
2       경상남도 거창군 월별 코로나19 관련 확진자 및 사망자 수 현황 데이터로 월별 확진...   
3       상기 데이터는 연도별 주민 1인당 또는 세대 당 부담된  지방세액에 대한 정보로 조...   
4       전라남도 보성군 담배소매인지정현황에 관한 공공데이터로 업소명, 지번주소, 도로명주소...   
...                                                   ...   
127046  구로구의 식품을 제조 또는 가공할 때 기호가치를 향상시키거나 영양가치를 높일목적의 ...   
127047  송파구의 컨벤션센터 등 대규모 관광 수요를 유발하는 국제회의를 개최할 수 있는 시설...   
127048  중구의 컨벤션센터 등 대규모 관광 수요를 유발하는 국제회의를 개최할 수 있는 시설을...   
127049  강남구의 컨벤션센터 등 대규모 관광 수요를 유발하는 국제회의를 개최할 수 있는 시설...   
127050  송파구의 숙박 및 휴게 등의 목적으로 이용하는 장소 또는 시설에서 비디오물 시청기자...   

                                                      url      data_type  \
0        https://www.data.go.kr/data/15106898/fileData.do            텍스트   
1        https://www.data.go.kr/data/15106993/fileData.do            텍스트   
2        https://www.data.go.kr/data/15098865/fileData.do            텍스트   
3        https://www.data.go.kr/data/15080007/fileData.do            텍스트   
4        https://www.data.go.kr/data/15035564/fileData.do            텍스트   
...                                                   ...            ...   
127046  https://data.seoul.go.kr/dataList/OA-18293/S/1...  SHEET,OpenAPI   
127047  https://data.seoul.go.kr/dataList/OA-16990/S/1...  SHEET,OpenAPI   
127048  https://data.seoul.go.kr/dataList/OA-16968/S/1...  SHEET,OpenAPI   
127049  https://data.seoul.go.kr/dataList/OA-16989/S/1...  SHEET,OpenAPI   
127050  https://data.seoul.go.kr/dataList/OA-17365/S/1...  SHEET,OpenAPI   

                      source   ori_label ori_source label  
0                   충청남도 당진시  산업·통상·중소기업    공공데이터포털  제조소비  
1                   전라남도 강진군          농림    공공데이터포털  식품건강  
2                   경상남도 거창군          보건    공공데이터포털  보건의료  
3                   충청남도 부여군    재정·세제·금융    공공데이터포털  경제금융  
4                   전라남도 보성군      일반공공행정    공공데이터포털  행정법률  
...                      ...         ...        ...   ...  
127046  구로구 스마트도시정책관 빅데이터담당관          보건    서울데이터광장  보건의료  
127047  송파구 스마트도시정책관 빅데이터담당관       문화/관광    서울데이터광장  문화관광  
127048   중구 스마트도시정책관 빅데이터담당관       문화/관광    서울데이터광장  문화관광  
127049  강남구 스마트도시정책관 빅데이터담당관       문화/관광    서울데이터광장  문화관광  
127050  송파구 스마트도시정책관 빅데이터담당관       문화/관광    서울데이터광장  문화관광  

[127051 rows x 8 columns]

### 팀에서 구분해놓은 13개 + 1개 (기타) = 14개 Label

제조소비, 식품건강, 보건의료, 경제금융, 행정법률, <br/>
사회복지, 환경기상, 교통물류, 문화관광, 재난안전, <br/>
주택토지, 교육과학, 통일외교안보, 기타 (한국어, 이미지, 텍스트 등)

In [11]:
total_df['label'].unique()

array(['제조소비', '식품건강', '보건의료', '경제금융', '행정법률', '사회복지', '환경기상', '교통물류',
       '문화관광', '재난안전', '주택토지', '교육과학', '통일외교안보', '기타'], dtype=object)

### 일부 데이터는 출처 (URL)가 겹치는 항목들도 존재
URL 값을 기준으로 unique한 것만 찾으면, 기존에 127,051개에서 94,805개로 변화

In [12]:
len(total_df['url'].unique())

94805

### URL이 겹치는 항목들에서 title, ori_label, label 데이터 확인
확인해보니, 데이터 제공처마다 유사한 데이터지만, 세부적으로는 label이 다른 경우들도 존재 (우리 label 기준도 다르게 나옴)

In [13]:
for i, _ in total_df.value_counts('url').head(30)[1:].iteritems():
    print(total_df[total_df['url'] == i][['title', 'ori_label', 'label']], '\n')

                                          title ori_label label
77401   2020년 4월 경기지역화폐 일반발행카드의 가맹점별 일간 결제금액 정보      경제금융  경제금융
121137  2020년 3월 경기지역화폐 정책발행카드의 가맹점별 일간 결제금액 정보      행정법률  행정법률
121144  2020년 4월 경기지역화폐 정책발행카드의 가맹점별 일간 결제금액 정보      행정법률  행정법률 

                                    title ori_label label
7341    인천광역시 남동구_개별공시지가 조사 추진일정_20220824      지역개발  주택토지
101336           인천광역시 남동구_개별공시지가 조사 추진일정      환경자원  환경기상 

                                       title ori_label label
31520  한국농수산식품유통공사_양재 화훼공판장 입주점포 정보_20210912        농림  식품건강
81230           한국농수산식품유통공사_양재 화훼공판장 입주점포 정보     농수산식품  식품건강 

                                    title ori_label label
31945  경상북도 경주시_동궁원홈페이지 검색키워드 정보_20210906        통신  교육과학
51885           경상북도 경주시_동궁원홈페이지 검색키워드 정보      교육과학  교육과학 

                                 title ori_label label
9696    인천광역시 남동구_환경법 위반업체 현황_20220811        환경  환경기상
101334           인천광역시 남동구_환경법 위반업체 현황      환경자원  환경기상 

                                 title

In [14]:
total_df.to_csv('total_data.csv', index=False)